x-pack/test/functional/page_objects/lens_page.ts

```diff
@@ -205,5 +220,61 @@ export function LensPageProvider({ getService, getPageObjects }: FtrProviderCont
       await PageObjects.header.waitUntilLoadingHasFinished();
       await testSubjects.missingOrFail('lnsApp_saveAndReturnButton');
     },
+    /**
+     * Gets label of dimension trigger in dimension panel
+     *
+     * @param dimension - the selector of the dimension
+     */
+    async getDimensionTriggerText(dimension: string, index = 0) {
+      const dimensionElements = await testSubjects.findAll(dimension);
+      const trigger = await testSubjects.findDescendant(
+        'lns-dimensionTrigger',
+        dimensionElements[index]
+      );
+      const text = await trigger.getVisibleText();
+      return text;
+    },
+
+    /**
+     * Gets text of the specified datatable header cell
+     *
+     * @param index - index of th element in datatable
+     */
+    async getDatatableThText(index = 0) {
+      return find
+        .byCssSelector(
+          `[data-test-subj="lnsDataTable"] thead th:nth-child(${
+            index + 1
+          }) .euiTableCellContent__text`
+        )
+        .then((el) => el.getVisibleText());
+    },
+
+    /**
+     * Gets text of the specified datatable cell
+     *
+     * @param rowIndex - index of row of the cell
+     * @param colIndex - index of column of the cell
```

Comment:
This looks brittle again, like I wouldn't want to reuse this function for other uses.

Reply:
How we were testing it before, was to check only the first element, so the index could have been not specified explicitly but it worked the same. I'm not exactly understanding what you mean by this being brittle and why you wouldn't reuse this function for other uses –  you're more than welcome to write the example.

Reply:
I was mostly expecting the selector to work backwards. Here, you provide the expected index and get the text at that index. Is there a way to provide the text, and get the index of the text?